src/com/stuypulse/stuylib/math/SLMath.java

```diff
@@ -145,6 +145,128 @@ public static double circular(double x) {
         return circular(x, 2);
     }
 
+    /*******************************/
+    /*** INTEGRAL APPROXIMATIONS ***/
+    /*******************************/
+
+    /**
+     * Given data points, calculates integral using the chosen formula.<br>
+     * 
+     * Arrays x and y must be the same length, because they contain data points.
+     * The x-values must be in sorted order.
+     * 
+     * @param x       the x-values of the data points, where (x[i], y[i]) is a point
+     * @param y       the y-values of the data points, where (x[i], y[i]) is a point
+     * @param formula the chosen formula to approximate the integral.
+     * @return approximation of the integral from given points
+     */
+    public static double integrate(double[] x, double[] y, Integral formula) {
+        // Precondition: there always exists x[i], y[i]
+        if(x.length != y.length) {
+            throw new IllegalArgumentException("x and y-value arrays should be the same length");
+        }
+
+        // Precondition: x-values should be sorted
+        for(int i = 1; i < x.length; ++i) {
+            if(x[i] < x[i - 1]) {
+                throw new IllegalArgumentException("x-values should be in sorted order");
+            }
+        }
+
+        return formula.calculate(x, y);
```

Comment:
I understand that the x values need to be sorted  but I think theres a better way